pkg/profile/artifact.go

```diff
@@ -6,28 +6,51 @@ import (
 	"strings"
 	"time"
 
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
 	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
-	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
-	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
 )
 
+// location defines where the helm chart is located.
+// remote -> in a helm repository
+// local -> in the profile repository
+type location int
+
+var local location = 0
+var remote location = 1
+
 // CreateArtifacts creates and inserts objects to the cluster to deploy the
 // profile as a HelmRelease.
 func (p *Profile) CreateArtifacts(ctx context.Context) error {
-	if err := p.createGitRepository(ctx); err != nil {
-		return fmt.Errorf("failed to create GitRepository resource: %w", err)
-	}
-
+	// TODO don't depend on 0.
 	switch kind := p.definition.Spec.Artifacts[0].Kind; kind {
-	case "HelmChart":
-		if err := p.createHelmRelease(ctx); err != nil {
+	case profilesv1.HelmChartLocalKind:
+		if err := p.createGitRepository(ctx); err != nil {
+			return fmt.Errorf("failed to create GitRepository resource: %w", err)
+		}
+		if err := p.createHelmRelease(ctx, local); err != nil {
 			return fmt.Errorf("failed to create HelmRelease resource: %w", err)
 		}
-	case "Kustomize":
+	case profilesv1.HelmChartRemoteKind:
+		if p.definition.Spec.Artifacts[0].HelmURL == "" {
+			return fmt.Errorf("helmURL for helm repository for remote helm artifact must not be empty")
+		}
+		//TODO: figure out what to do with the chart name and version. Maybe they could be inferred?
```

Comment:
Can you expand a bit more on this? can't you use `HelmChartVersion` ?

Reply:
I asked in this Slack. It's more like a conversation about how deep we want to surface Helm Chart values in profile yaml data. At some point it ought to be too much.